blockchain/v2/scheduler.go

```diff
@@ -383,3 +469,107 @@ func (sc *schedule) numBlockInState(targetState blockState) uint32 {
 	}
 	return num
 }
+
+// Handlers
+
+// XXX: This handler will probably get the block first and then pass it on to the processor
+func (sc *scheduler) handleBlockResponse(event tmpBlockResponse) (Event, error) {
+	err := sc.touchPeer(event.peerID, event.time)
+	if err != nil {
+		return scPeerError{peerID: event.peerID, reason: err}, nil
+	}
+
+	err = sc.markReceived(event.peerID, event.block.Height, event.size, event.time)
+	if err != nil {
+		return scPeerError{peerID: event.peerID, reason: err}, nil
+	}
+
+	return scBlockReceived{peerID: event.peerID, block: event.block}, nil
+}
+
+func (sc *scheduler) handleBlockProcessed(event pcBlockProcessed) (Event, error) {
+	err := sc.markProcessed(event.height)
+	if err != nil {
+		return scSchedulerFail{reason: err}, nil // this should be fatal
+	}
+
+	if sc.allBlocksProcessed() {
+		return scFinishedEv{}, nil
+	}
+
+	return noOp, nil
+}
+
+// XXX: unify types peerError
+func (sc *scheduler) handlePeerError(event peerError) (Event, error) {
+	err := sc.removePeer(event.peerID)
+	if err != nil {
+		return scSchedulerFail{reason: err}, nil // xxx: bit extreme
+	}
+
+	return noOp, nil
+}
+
+func (sc *scheduler) handleTryPrunePeer(event tryPrunePeer) (Event, error) {
```

Comment:
Add some comments. This removes only one peer, not sure why. Could return an event with a list of pruned peers.

Reply:
changed to return list of pruned peers.